tests/lib/prepare.sh

```diff
@@ -2,6 +2,41 @@
 
 set -eux
 
+update_core_snap_with_snap_exec_snapctl() {
+    # We want to use the in-tree snap-exec and snapctl, not the ones in the core
+    # snap. To accomplish that, we'll just unpack the core we just grabbed,
+    # shove the new snap-exec and snapctl in there, and repack it.
+
+    # First of all, unmount the core
+    core="$(realpath /snap/ubuntu-core/current)"
+    snap="$(mount | grep "$core" | awk '{print $1}')"
```

Comment:
If you change this line to `snap="$(mount | grep " $core" | awk '{print $1}')"` (notice the space before `$core`) it'll work on ubuntu-core.